crates/oracle/src/subgraph.rs

```diff
@@ -207,3 +255,33 @@ where
 
     de.deserialize_string(HexStringVisitor)
 }
+
+mod helpers {
+    use super::*;
+    pub fn duplicates<I, J>(elements: I) -> HashSet<J>
+    where
+        I: IntoIterator<Item = J>,
+        J: Ord + Eq + Clone + std::hash::Hash,
+    {
+        let sorted = elements.into_iter().sorted();
+        let mut duplicates = HashSet::new();
+        for (a, b) in sorted.tuple_windows() {
+            if a == b {
+                duplicates.insert(a);
+            }
+        }
+        duplicates
+    }
```

Comment:
There's https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.duplicates for this :smile:

Reply:
Thanks! Not my smartest move :smile: 
Fixed in 75148f3